src/pages/ThorChainLP/components/RemoveLiquidity/RemoveLiquidityInput.tsx

```diff
@@ -178,43 +251,232 @@ export const RemoveLiquidityInput: React.FC<RemoveLiquidityProps> = ({
+      if (
+        !actualRuneCryptoLiquidityAmount ||
+        !actualAssetCryptoLiquidityAmount ||
+        !poolAsset ||
+        !userData ||
+        !poolAssetUserlpData
+      )
+        return
+
+      const runeAmountCryptoThorPrecision = convertPrecision({
+        value: actualRuneCryptoLiquidityAmount,
+        inputExponent: 0,
+        outputExponent: THOR_PRECISION,
+      }).toFixed()
+
+      const assetAmountCryptoThorPrecision = convertPrecision({
+        value: actualAssetCryptoLiquidityAmount,
+        inputExponent: 0,
+        outputExponent: THOR_PRECISION,
+      }).toFixed()
+
+      setIsSlippageLoading(true)
+
+      const estimate = await estimateRemoveThorchainLiquidityPosition({
+        userData: poolAssetUserlpData,
+        assetId: poolAsset.assetId,
+        runeAmountCryptoThorPrecision,
+        assetAmountCryptoThorPrecision,
+      })
+
+      console.log('xxx debug estimate', { estimate })
+
+      setIsSlippageLoading(false)
+
+      setSlippageRune(
+        bnOrZero(estimate.slipPercent)
+          .div(100)
+          .times(virtualRuneFiatLiquidityAmount ?? 0)
+          .times(2)
+          .toFixed(),
+      )
+    })()
+  }, [
+    actualAssetCryptoLiquidityAmount,
+    actualRuneCryptoLiquidityAmount,
+    actualRuneFiatLiquidityAmount,
+    poolAsset,
+    foundPool?.asymSide,
+    foundPool?.isAsymmetric,
+    isAsym,
+    isAsymAssetSide,
+    isAsymRuneSide,
+    virtualRuneFiatLiquidityAmount,
+    userData,
+    poolAssetUserlpData,
+  ])
+
   const tradeAssetInputs = useMemo(() => {
-    if (!(asset && rune && foundPool)) return null
+    if (!(poolAsset && rune && foundPool)) return null
 
     const assets: Asset[] = (() => {
-      switch (foundPool.asymSide) {
-        case null:
-          return [rune, asset]
-        case AsymSide.Rune:
-          return [rune]
-        case AsymSide.Asset:
-          return [asset]
-        default:
-          assertUnreachable(foundPool.asymSide)
-      }
+      if (foundPool.asymSide === null) return [poolAsset, rune]
+      if (foundPool.asymSide === AsymSide.Rune) return [rune]
+      if (foundPool.asymSide === AsymSide.Asset) return [poolAsset]
+
+      throw new Error('Invalid asym side')
     })()
 
     return (
       <Stack divider={pairDivider} spacing={0}>
-        {assets.map(_asset => {
+        {assets.map(asset => {
+          const isRune = asset.assetId === rune.assetId
+          const marketData = isRune ? runeMarketData : assetMarketData
+          const handleRemoveLiquidityInputChange = createHandleRemoveLiquidityInputChange(
+            marketData,
+            isRune,
+          )
+          const cryptoAmount = isRune
+            ? virtualRuneCryptoLiquidityAmount
+            : virtualAssetCryptoLiquidityAmount
+          const fiatAmount = isRune
+            ? virtualRuneFiatLiquidityAmount
+            : virtualAssetFiatLiquidityAmount
+
+          // const accountId = accountIdsByChainId[asset.chainId]
```

Comment:
Was going to comment on this one but remembered this was noted as a TODO in the description ✅